core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -815,8 +848,9 @@ class AuthorizerIntegrationTest extends BaseRequestTest {
         assertEquals(Collections.singleton(newTopic), e.unauthorizedTopics())
     }
 
-    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), newTopicResource)
-    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+    // using topic-level create Acl in this test, leaving cluster-level create Acl in the matching `write` test
+    addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write), new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)),
+      newTopicResource)
```

Comment:
It's safer to have asserts for both cases here and in the other place.

Reply:
Ok, added test cases for all code paths